src/py42/services/savedsearch.py

```diff
@@ -61,3 +62,20 @@ def execute(self, search_id, page_number=None, page_size=None):
         """
         query = self.get_query(search_id, page_number=page_number, page_size=page_size)
         return self._file_event_client.search(query)
+
+    def execute_get_all(self, search_id, page_token=""):
```

Comment:
I feel `execute_pagination` might be a bit better, `get_all` feels like a call once and done method or returning a stream that I interact with over having to call again with the next token.

Reply:
Are we avoiding changing `execute()` to use pagination to not break anything for existing usage? If so then I think keeping the name `execute_get_all()` might make more sense to really drive home the difference between the methods.

Reply:
@ryan-haley-code42 I don't think it'd be a breaking change. However, in the `securitydata` client, there are two methods `search_file_events()` and `search_all_file_events()` that have this same distinction (one support `page_token` and one does not.

Also, I should add that both technically support pagination. There are like two levels of pagination when it comes to these file event search APIs... It is just that they originally caps the number of events to 10,000 to prevent API abuse. However, we had to write scripts to get around this limitation because customers needed all the events... Thus, they added the ability to search past 10,000 via this page token.

I am not sure exactly the motive behind why there are two methods `search_file_events()/search_all_file_events()`, but I remember the original plan being that the `search_all()` one would return a generator.  It appears that is not the case. However, methods in py42 that return generators _should_ include the keyword `all` somewhere in the method name.  I suppose these file event search methods are different, but we seem to have done the same distinction via the `all` keyword.

I think it is important, given that savedsearches are completely paired with the searching file event API, that it mirrors the same method distinction of the one having `page_token` having the word `all` somewhere in the method name.

That being said,  this is a difficult method to name and I don't really have a better idea, I think `execute_get_all()` would work but am curious what others think.

Sorry, that is a lot of info

Reply:
If we go this route, someone should remember to add a ticket in the py42 2.0 GitHub project that says to remove the `execute` methods